Rasa_Bot/actions/actions_general_activity.py

```diff
@@ -496,3 +504,255 @@ def get_random_activities(avoid_activity_id: int, number_of_activities: int):
+
+
+class SendPersuasiveMessageActivity(Action):
+    """Choose persuasion type and send persuasive message (if any)"""
+
+    def name(self):
+        return "send_persuasive_message_activity"
+
+    async def run(self, dispatcher, tracker, domain):
+        
+        chosen_option = int(tracker.get_slot('general_activity_next_activity_slot'))
+        activities_slot = tracker.get_slot('rnd_activities_ids')
+
+        activity_id = activities_slot[chosen_option - 1]
+        session = get_db_session(db_url=DATABASE_URL)
+
+        # Get the activity benefit
+        activities = (
+            session.query(
+                InterventionActivity
+            )
+            .filter(
+                InterventionActivity.intervention_activity_id == activity_id
+            ).all()
+        )
+        benefit = activities[0].intervention_activity_benefit
+        
+        # Get optimal persuasion type for user state
+        pers_type = get_opt_persuasion_type(tracker)
+        
+        reflective_question = ""  # Reflective question to ask users as part of persuasion
+        input_required = False  # Whether we will ask a reflective question to users
+        message_idx = -1  # Index of persuasive message
+        # Commitment
+        if pers_type == 0:
+            input_required = True
+            message_idx = random.choice(list(range(len(COMMITMENT))))
+            dispatcher.utter_message(text=COMMITMENT[message_idx])
+            # Not identity-based formulation
+            if message_idx < 4:
+                reflective_question = REFLECTIVE_QUESTION_COMMITMENT
+            # Identity-based formulation
+            else:
+                reflective_question = REFLECTIVE_QUESTION_COMMITMENT_IDENTITY
+        # Consensus
+        elif pers_type == 1:
+            input_required = True
+            message_idx = random.choice(list(range(len(CONSENSUS))))
+            dispatcher.utter_message(text=CONSENSUS[message_idx] + " " + benefit)
+            reflective_question = REFLECTIVE_QUESTION_CONSENSUS
+        
+        return [SlotSet("persuasion_type", pers_type),
+                SlotSet("persuasive_message_index", message_idx),
+                SlotSet("persuasion_requires_input", input_required),
+                SlotSet("persuasion_reflective_question", reflective_question)]
+    
+
+class ValidatePersuasionWantForm(FormValidationAction):
+    def name(self) -> Text:
+        return 'validate_persuasion_want_form'
+
+    def validate_persuasion_want_slot(
+            self, value: Text, dispatcher: CollectingDispatcher,
+            tracker: Tracker, domain: Dict[Text, Any]) -> Dict[Text, Any]:
+        # pylint: disable=unused-argument
+        """Validate persuasion_want_slot input."""
+        last_utterance = get_latest_bot_utterance(tracker.events)
+
+        if last_utterance != 'utter_ask_persuasion_want_slot':
+            return {"persuasion_want_slot": None}
+
+        if not self._validate_response_value(value):
```

Comment:
After importing the `validator.py`
```suggestion
        if not validator.validate_number_in_range_response(1, 5, value):
```